### PR TITLE
Fix performance issues

### LINKED ISSUE #148: Figure out and fix memory leak
Not necessarily introduced by #44, but it certainly exacerbated the issue. Before base memory usage (on the settings page) was ~80MiB, now it is around ~800MiB.

### PR DESCRIPTION
**Pull request by robbevp**
fix #148 #154

### CODE REVIEW

**Comment by robbevp** [comment]
Todo:
* [ ] convert `setObject` mutation to new system
* [ ] convert `updateOccurences` to replace whole object

**Comment by robbevp** [comment]
@charvp I've tested this with some edits for various types of objects. There is a small spike in memory usage after a mass edit (tested with 40 tracks), but the garbage collector is able to handle this after a while.

**Comment by charvp** [comment]
@robbevp Can you take a look at my commit. I'll approve and merge once I've heard something from you.

File: src/store/albums.js [hunk 1]
```diff
@@ -44,21 +44,19 @@ export default {
         }
       }
     },
-    updateLabelOccurence(state, { newID, oldID }) {
-      for (const a in state.albums) {
-        const i = state.albums[a].label_ids.findIndex(
-          (lId) => `${lId}` === `${oldID}`
-        );
+    updateLabelOccurence(state, { oldID, newID }) {
+      const oldAlbums = state.albums;
```
Comment (robbevp) [inline src/store/albums.js:48]: Is there a reason why this doesn't use the spread operator? I would want to keep all the commit as similar as possible

Comment (robbevp) [reply src/store/albums.js:48]: If it isn't necessary to use a spread operator, we should remove it for all `const oldObjects = `

Comment (charvp) [reply src/store/albums.js:48]: Because it's not necessary? No need to create a new (large) object if it's not required.

Comment (charvp) [reply src/store/albums.js:48]: > If it isn't necessary to use a spread operator, we should remove it for all `const oldObjects = `

Sure, agreed.

**Pull request merged**